Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,398 @@
+#include <Prefab/ViewBookmarkLoader.h>
+
+/*!
+ * ViewBookmarkIntereface
+ * Interface for saving/loading View Bookmarks.
+ */
+namespace AzToolsFramework
+{
+    namespace Prefab
+    {
+        static constexpr const char* s_viewBookmarksRegistryPath = "/O3DE/ViewBookmarks/";
+
+        void ViewBookmarkLoader::RegisterViewBookmarkLoaderInterface()
+        {
+            AZ::Interface<ViewBookmarkLoaderInterface>::Register(this);
+        }
+
+        void ViewBookmarkLoader::UnregisterViewBookmarkLoaderInterface()
+        {
+            AZ::Interface<ViewBookmarkLoaderInterface>::Unregister(this);
+        }
+
+        void ViewBookmarkLoader::SaveBookmarkSettingsFile()
+        {
+            auto registry = AZ::SettingsRegistry::Get();
+            if (registry == nullptr)
+            {
+                AZ_Warning("SEditorSettings", false, "Unable to access global settings registry. Editor Preferences cannot be saved");
+                return;
+            }
+
+            // Resolve path to user folder
+            AZ::IO::FixedMaxPath editorBookmarkFilePath =
+                AZ::IO::FixedMaxPath(AZ::Utils::GetProjectPath()) / "user/Registry/ViewBookmarks/";
+
+            editorBookmarkFilePath /= m_bookmarkfileName;
+
+            AZ::SettingsRegistryMergeUtils::DumperSettings dumperSettings;
+            dumperSettings.m_prettifyOutput = true;
+
+            AZStd::string bookmarkKey = "/" + m_bookmarkfileName;
+            dumperSettings.m_includeFilter = [&bookmarkKey](AZStd::string_view path)
+            {
+                AZStd::string_view o3dePrefixPath(bookmarkKey);
+                return o3dePrefixPath.starts_with(path.substr(0, o3dePrefixPath.size()));
+            };
+
+            AZStd::string stringBuffer;
+            AZ::IO::ByteContainerStream stringStream(&stringBuffer);
+            if (!AZ::SettingsRegistryMergeUtils::DumpSettingsRegistryToStream(*registry, "", stringStream, dumperSettings))
+            {
+                AZ_Warning(
+                    "SEditorSettings", false, R"(Unable to save changes to the Editor Preferences registry file at "%s"\n)",
+                    editorBookmarkFilePath.c_str());
+                return;
+            }
+
+            bool saved{};
+            constexpr auto configurationMode =
+                AZ::IO::SystemFile::SF_OPEN_CREATE | AZ::IO::SystemFile::SF_OPEN_CREATE_PATH | AZ::IO::SystemFile::SF_OPEN_WRITE_ONLY;
+            if (AZ::IO::SystemFile outputFile; outputFile.Open(editorBookmarkFilePath.c_str(), configurationMode))
+            {
+                saved = outputFile.Write(stringBuffer.data(), stringBuffer.size()) == stringBuffer.size();
+            }
+
+            // Once written to the desired file remove the key from the settings registry
+            registry->Remove(bookmarkKey + "/");
+            AZ_Warning(
+                "SEditorSettings", saved, R"(Unable to save Editor Preferences registry file to path "%s"\n)",
```

Comment:
Repeat: This is referring to the Editor Preferences and not the Bookmark settings registry file